### ISSUE TITLE
Clear button disabled when there is no other way to unset a first day

### ISSUE THREAD

**Issue by JordyGit** [benitolopez/hotel-datepicker#120]
I'm creating a date picker where only the available ranges are possible to be chosen.
I will disable days where an invalid range is not possible but out of the box there are combinations possible where you can select a last day but not a first day and the clear button doesn't show up yet when only the first/last day is selected. Only possible way to get out of it is by refreshing the page. 

Fix: allow to clear the moment the first/last day is chosen

Workaround: refresh page

Gif, all works except the first range because the first day can't be chosen after selecting the last day, no clear button is enabled:
![Animatie](https://user-images.githubusercontent.com/33500507/182462981-c49c99d1-26a3-40d7-999c-9d85bd8179e9.gif)
*some css was adjusted to remove the red backgrounds etc

Code:
```
var rangePicker = new HotelDatepicker(document.getElementById('range-input-parent'), {
		inline: true,
		clearButton: true,
		moveBothMonths: true,
		selectForward: false,
		startOfWeek: 'monday',
		format: 'DD/MM/YYYY',
		minNights: 6,
		maxNights: 6,
		noCheckInDates: [
			'2022-08-02',
			'2022-08-03',
			'2022-08-04',
			'2022-08-05',
			'2022-08-06',
			//'2022-08-07',
			//'2022-08-08',
			'2022-08-09',
			'2022-08-10',
			'2022-08-11',
			'2022-08-12',
			'2022-08-13',
			//'2022-08-14',
			// '2022-08-15',
			'2022-08-16',
			'2022-08-17',
			'2022-08-18',
			'2022-08-19',
			'2022-08-20',
			//'2022-08-21',
			// '2022-08-22',
			'2022-08-23',
			'2022-08-24',
			'2022-08-25',
			'2022-08-26',
			'2022-08-27',
			//'2022-08-28',
			//
		],
		noCheckOutDates: [
			//'2022-08-02',
			'2022-08-03',
			'2022-08-04',
			'2022-08-05',
			'2022-08-06',
			//'2022-08-07',
			//'2022-08-08',
			'2022-08-09',
			'2022-08-10',
			'2022-08-11',
			'2022-08-12',
			'2022-08-13',
			// '2022-08-14',
			//'2022-08-15',
			'2022-08-16',
			'2022-08-17',
			'2022-08-18',
			'2022-08-19',
			'2022-08-20',
			// '2022-08-21',
			//'2022-08-22',
			'2022-08-23',
			'2022-08-24',
			'2022-08-25',
			'2022-08-26',
			'2022-08-27',
			// '2022-08-28',
		],
		disabledDates: [
			// '2022-08-13'
		]
	});
```

**Comment by benitolopez** [comment]
Hello,

Thank you for the detailed report. Yes, you're right.

Fixed in 4.3.1 (https://github.com/benitolopez/hotel-datepicker/releases/tag/4.3.1)

**Comment by JordyGit** [comment]
Nice! Happy to see something is done with my input ;)
